### PR TITLE
Fulfilling customer request for ToString return information about errorDetails

### PR DESCRIPTION
**Pull request by pgilmorepf**
GenerateErrorReport returns all information about the error.
ToString was providing a streamlined set of information, but that was confusing to customers, so now ToString just calls GetErrorReport.
Ancillary changes:
Separating GetServerTime from UserDataApi, so that each test is distinct and less complicated
Fixing a few typos and misspellings.

### CODE REVIEW

**Comment by toddbellMSFT** [comment]
Did you explicitly want to edit the meta png file? Seems like excess edits. 

Based on the title of the PR, I assumed I was just going to look at the last file. It took me a sec when I was reading tests for Timestamps which seems to be similar amount of work. I'll sign off on both, but I'd prefer to see them as separate micro PR's for cleaner git commit histories

File: targets/unity-v2/Testing/Tests/Client/ClientApiTests.cs [hunk 1]
```diff
@@ -254,14 +257,31 @@ private void GetUserDataCallback2(GetUserDataResult result)
                 int.TryParse(userDataRecord.Value, out actualValue);
             testContext.IntEquals(_testInteger, actualValue);
             testContext.NotNull(userDataRecord, "UserData record not found");
+            testContext.EndTest(UUnitFinishState.PASSED, actualValue.ToString());
+        }
+
+        /// <summary>
+        /// CLIENT API
+        /// Test that the DateTime fields deserialize correctly, and accurately represent UTC time
+        /// Relies on the local machine time and time zone being set correctly.
```
Comment (toddbellMSFT) [inline targets/unity-v2/Testing/Tests/Client/ClientApiTests.cs]: What does it mean to set the machine correctly? How will this work on our VMs? (I'm assuming we expect them to default to the time zone based on the data center?)

File: targets/unity-v2/Testing/Tests/Client/ClientApiTests.cs [hunk 2]
```diff
@@ -254,14 +257,31 @@ private void GetUserDataCallback2(GetUserDataResult result)
...
-            var timeUpdated = userDataRecord.LastUpdated;
+            var serverTime = result.Time;
             var minTest = DateTime.UtcNow - TimeSpan.FromMinutes(5);
             var maxTest = DateTime.UtcNow + TimeSpan.FromMinutes(5);
-            testContext.True(minTest <= timeUpdated && timeUpdated <= maxTest);
+            testContext.True(minTest <= serverTime && serverTime <= maxTest);
 
-            testContext.True(Math.Abs((DateTime.UtcNow - timeUpdated).TotalMinutes) < 5); // Make sure that this timestamp is recent - This must also account for the difference between local machine time and server time
-            testContext.EndTest(UUnitFinishState.PASSED, actualValue.ToString());
+            testContext.True(Math.Abs((DateTime.UtcNow - serverTime).TotalMinutes) < 5); // Make sure that this timestamp is recent - This must also account for the difference between local machine time and server time
```
Comment (toddbellMSFT) [inline targets/unity-v2/Testing/Tests/Client/ClientApiTests.cs]: Our documentation for [GetTime](https://api.playfab.com/documentation/Client/method/GetTime) seems to indicate we can have a potential variance of up to 5 seconds. Should this test be in minutes? or Should we expect/test stricter time limits?

Comment (pgilmorepf) [reply targets/unity-v2/Testing/Tests/Client/ClientApiTests.cs]: The Server may have a variance of 5 minutes, but by experimentation I can tell you that cloud instances themselves have upto a ~2 minute variance in time, and that most local machines have the same variance. In fact, there's still a rare but plausible case where 5 minutes isn't even enough, but in practice it seems to be >99% stable.

File: targets/unity-v2/Testing/Tests/Client/ClientApiTests.cs (hunk 1 above)
Comment (pgilmorepf) [reply targets/unity-v2/Testing/Tests/Client/ClientApiTests.cs]: Basically to have a clock that is accurately synced to internet-time, within the given threshold, of 5 minutes. In my experience, if you look at a dozen or more machines, one machine can deviate from an authoritative clock by upto ~2 minutes, even if it's set "properly".

**Comment by pgilmorepf** [comment]
I'm gonna admin land this.
The tests for Unity Windows passed, and some of the console specific tests are still offline, and therefore can't finish. These aren't changes that would affect consoles.

**Pull request merged**